### ISSUE TITLE
use of `from __future__ import annotations` makes pyright not raise errors for specific python versions

### ISSUE THREAD

**Issue by Popkornium18** [microsoft/pyright#10655]
Lets say we are linting this file:

```python
a: list[str] = []
```

When we lint it with `pyright --pythonversion=3.7 somefile.py` we get the expected error:

```
/path/to/somefile.py
  /path/to/somefile.py:2:4 - error: Subscript for class "list" will generate runtime exception; enclose type expression in quotes (reportIndexIssue)
1 error, 0 warnings, 0 informations
```

But if we modify the file to include `from __future__ import annotations` at the top of the file we get no error at all.

```
pyright --pythonversion=3.7 somefile.py
0 errors, 0 warnings, 0 informations
pyright --pythonversion=3.6 somefile.py
0 errors, 0 warnings, 0 informations
```

And we even get no errors in Python 3.6 as well! This is even worse, since in 3.6 even the `from __future__ import annotations` line itself should raise an error in pyright.

**Comment by erictraut** [comment]
Pyright relies on typeshed stubs for stdlib type information including much of the details about which features are available in each version of Pyright. Typeshed dropped support for Python 3.6 a few years ago. That explains why you don't see an error when you attempt to use `from __future__ import annotations` on Python 3.6. In fact, typeshed recently dropped support for Python 3.8. If you're still on 3.6 or 3.7, I recommend upgrading to at least 3.9 if that's practical.

You might be able to unearth an old copy of the typeshed stubs and configure pyright to use them.

On Python 3.7 and newer, `from __future__ import annotations` is supported. When this feature is in effect, annotation evaluations are not evaluated at runtime (or more accurately, are evaluated as though they are string literals), so you shouldn't see a runtime error if you use `list[str]` in an annotation on Python 3.7 with `from __future__ import annotations` in effect. That's why pyright doesn't generate an error in this case.

In any event, pyright is working as intended here, so I'm going to close the issue.
